pip/download.py

```diff
@@ -369,12 +371,28 @@ def __init__(self, *args, **kwargs):
         for host in insecure_hosts:
             self.mount("https://{0}/".format(host), insecure_adapter)
 
-    def request(self, method, url, *args, **kwargs):
+    def request(self, *args, **kwargs):
         # Allow setting a default timeout on a session
         kwargs.setdefault("timeout", self.timeout)
 
+        # If we're running inside of a zipimport, and we were not given an
+        # explicit certificate, then we want to unzip our bundled certificate.
+        if (not isinstance(self.verify, six.string_types) and
+                isinstance(
+                    getattr(requests, "__loader__", None),
+                    zipimport.zipimporter)):
+            cert_data = pkgutil.get_data("pip._vendor.requests", "cacert.pem")
+            if cert_data is not None:
+                with tempfile.NamedTemporaryFile(delete=False) as temp_cert:
```

Comment:
The tempfile module should do that yes:

> Creates a temporary file in the most secure manner possible. There are no race conditions in the file’s creation, assuming that the platform properly implements the os.O_EXCL flag for os.open(). The file is readable and writable only by the creating user ID. If the platform uses permission bits to indicate whether a file is executable, the file is executable by no one.

In addition, I'm not sure that I can use ``temp_cert.flush()`` here, because the tempfile module says:

> Whether the name can be used to open the file a second time, while the named temporary file is still open, varies across platforms (it can be so used on Unix; it cannot on Windows NT or later).

I don't have a Windows machine handy to test it, but I assume the documentation is correct and I have to close the file before it can be opened again by OpenSSL on Windows?